### ISSUE TITLE
ifshort false positive

### ISSUE THREAD

**Issue by gucio321** [golangci/golangci-lint#2662]
### Welcome

- [X] Yes, I'm using a binary release within 2 latest major releases. Only such installations are supported.
- [X] Yes, I've searched similar issues on GitHub and didn't find any.
- [X] Yes, I've included all information below (version, config, etc).
- [X] Yes, I've tried with the standalone linter if available. (https://golangci-lint.run/usage/linters/)

### Description of the problem

ifshort linter returns false positive

### Version of golangci-lint

<details>

```console
$ golangci-lint --version
golangci-lint has version 1.45.0 built from 1f4c1ed9 on 2022-03-18T15:08:39Z
```

</details>

### Configuration file

it is easier to reproduce with:
```sh
golangci-lint run --disable-all --enable ifshort ./...
```

### Go environment

<details>

```console
$ go version && go env
go version go1.18 linux/amd64
GO111MODULE=""
GOARCH="amd64"
GOBIN=""
GOCACHE="/home/me/.cache/go-build"
GOENV="/home/me/.config/go/env"
GOEXE=""
GOEXPERIMENT=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="linux"
GOINSECURE=""
GOMODCACHE="/home/me/go/pkg/mod"
GONOPROXY=""
GONOSUMDB=""
GOOS="linux"
GOPATH="/home/me/go"
GOPRIVATE=""
GOPROXY="https://proxy.golang.org,direct"
GOROOT="/usr/local/go"
GOSUMDB="sum.golang.org"
GOTMPDIR=""
GOTOOLDIR="/usr/local/go/pkg/tool/linux_amd64"
GOVCS=""
GOVERSION="go1.18"
GCCGO="gccgo"
GOAMD64="v1"
AR="ar"
CC="gcc"
CXX="g++"
CGO_ENABLED="1"
GOMOD="/dev/null"
GOWORK=""
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -fmessage-length=0 -fdebug-prefix-map=/tmp/go-build879183099=/tmp/go-build -gno-record-gcc-switches"
```

</details>

### Verbose output of running

<details>

```console
$ golangci-lint cache clean
$ golangci-lint run -v
INFO [config_reader] Config search paths: [./ /home/mszeptuch/dbg /home/mszeptuch /home /] 
INFO [lintersdb] Active 1 linters: [ifshort]      
INFO [loader] Go packages loading at mode 7 (files|name|compiled_files) took 119.459984ms 
INFO [runner/filename_unadjuster] Pre-built 0 adjustments in 118.164µs 
INFO [linters context/goanalysis] analyzers took 0s with no stages 
INFO [runner] Processors filtering stat (out/in): autogenerated_exclude: 1/1, uniq_by_line: 1/1, diff: 1/1, path_shortener: 1/1, cgo: 1/1, skip_files: 1/1, skip_dirs: 1/1, severity-rules: 1/1, sort_results: 1/1, filename_unadjuster: 1/1, max_same_issues: 1/1, source_code: 1/1, max_per_file_from_linter: 1/1, path_prefixer: 1/1, path_prettifier: 1/1, identifier_marker: 1/1, exclude: 1/1, exclude-rules: 1/1, nolint: 1/1, max_from_linter: 1/1 
INFO [runner] processing took 148.999µs with stages: nolint: 36.741µs, exclude-rules: 33.504µs, identifier_marker: 31.377µs, autogenerated_exclude: 15.224µs, path_prettifier: 13.29µs, skip_dirs: 6.448µs, source_code: 5.966µs, uniq_by_line: 1.736µs, path_shortener: 1.217µs, max_same_issues: 1.044µs, cgo: 552ns, max_from_linter: 466ns, filename_unadjuster: 437ns, skip_files: 269ns, max_per_file_from_linter: 230ns, exclude: 174ns, sort_results: 96ns, diff: 85ns, severity-rules: 72ns, path_prefixer: 71ns 
INFO [runner] linters took 499.292µs with stages: ifshort: 310.064µs 
main.go:30:2: variable 'state' is only used in the if-statement (main.go:31:2); consider using short syntax (ifshort)
	state := getState()
	^
INFO File cache stats: 1 entries of total size 495B 
INFO Memory: 3 samples, avg is 50.6MB, max is 50.6MB 
INFO Execution took 122.969731ms
```

</details>

### Code example or link to a public repository

<details>

```go
package main

import (
	"fmt"

	"github.com/AllenDang/giu"
)

type state struct {
	x int32
}

func getState() *state {
	return &state{}
}

func loop() {
	giu.SingleWindow().Layout(
		Widget(),
	)
}

type widget struct{}

func Widget() *widget {
	return &widget{}
}

func (w *widget) Build() {
// here is the exact point
// <------------------------------>
	state := getState()
	if state.x == 1 {
		fmt.Println("it is 1")
	}

	giu.Layout{
		giu.InputInt(&state.x),
	}.Build()
// <------------------------------>
}

func main() {
	wnd := giu.NewMasterWindow("test", 640, 480, 0)
	wnd.Run(loop)
}
```

</details>

**Comment by bombsimon** [comment]
You ticked the last checkbox that says
> Yes, I've tried with the standalone linter if available. (https://golangci-lint.run/usage/linters/)

So I guess you saw this was a issue with the `ifshort` linter and not `golangci-lint`. Based on that the best thing to do is to file an upstream issue at https://github.com/esimonov/ifshort/issues (if not already there, there are quite a few false positives).

Here's a shorter repro without dependencies and the result:

**Repro**
<details>

```go
package main

import (
        "fmt"
)

type state struct {
        x int32
}

type almostInt int32

type other struct {
        almostInt
}

func getState() *state {
        return &state{}
}

func main() {
        state := getState()
        if state.x == 1 {
                fmt.Println("it is 1")
        }

        _ = other{
                almostInt(state.x),
        }
}
```

</details>

```sh
$ golangci-lint run --no-config --disable-all --enable ifshort --print-issued-lines=false  ./...
main.go:22:2: variable 'state' is only used in the if-statement (main.go:23:2); consider using short syntax (ifshort)
```

```sh
$ ifshort ./
/tmp/x/main.go:22:2: variable 'state' is only used in the if-statement (/tmp/x/main.go:23:2); consider using short syntax
```

**Comment by ldez** [comment]
Close in favor of the issue on ifshort repository.
